### ISSUE TITLE
Rely on app config file

### ISSUE THREAD

**Issue by SBoudrias** [rmurphey/js-testing-boilerplates#11]
Hi, right now in the [`config.js`](https://github.com/rmurphey/js-testing-boilerplates/blob/master/mocha-boilerplate/config.js) file in the test folders, we are somehow reconfiguring most of our libs and shim that we probably already set up (with `requirejs`) in our app folder.

I wonder if there would be a way to leverage this to use multiple configurations and prevent config repetition. Maybe a bit like [`Backbone boilerplate`](https://github.com/tbranyen/backbone-boilerplate/blob/master/test/jasmine/index.html#L25) did with their jasmine and qunit tests. (Although I think there's a problem in backbone-boilerplate as it didn't wait for the first asynchronous config to be loaded to start the test suite)

Any though on this?

**Comment by rmurphey** [comment]
I agree that this is an issue -- the idea is that I didn't want to force people to buy into require/AMD for their project, while still taking advantage of them for the tests. But I agree that currently, there's not great support for people who _have_ bought into require/AMD in their main project. I'm not sure whether to address this just in the readme, or to provide more elaborate support for it via code. What do you think?

**Comment by SBoudrias** [comment]
I'm personally under the impression that most developper testing their app are working on some complex application where AMD is used most of the time (or at least commonjs modules). So, I may have take this the other way around, default with AMD and example in readme to use without... Although, you could also only create two folder `mocha-AMD`, `mocha`.

Do you have any working example of Mocha test using AMD and grunt? I've been able to set it up pretty easily in the browser, but I keep facing a `phantomjs timeout` using `grunt-mocha`. (Though I haven't looked really deep in this bug yet)

**Comment by SBoudrias** [comment]
What do you think about setting up two _demo app_ - one AMD, one without. And then show multiple configuration to work with both style?

```
- demo-app
- demo-AMD-app
- mocha-boilerplate
- mocha-AMD-boilerplate
```

I've been working on an example AMD/mocha test here: https://github.com/SBoudrias/grunt-mocha-example
I'd need to clean up some stuff, but if it's good for you I could send a pull request with it.
